src/contract/table-cursor.ts

```diff
@@ -94,21 +99,30 @@ export class TableCursor<TableRow> {
             indexPosition = fieldToIndexMapping[this.indexPositionField].index_position
         }
 
-        const {rows, next_key} = await this.table.contract.client!.v1.chain.get_table_rows({
+        const result = await this.table.contract.client!.v1.chain.get_table_rows({
             ...this.tableParams,
-            limit: Math.min(this.tableParams.limit - this.rowsCount, 1000000),
+            limit: Math.min(this.maxRows - this.rowsCount, this.tableParams.limit),
```

Comment:
I don't think this is gonna work when a large `limit` value is passed. In that situation, I think that you would want to also subtract `rowsCount` from `this.tableParams.limit`.

Reply:
If you can, write up some failing tests for situations where this might fail. 

AFAIK I think it'll work fine? The point of this limit is just to ensure we're not querying more than we need, and in this situation we're either going to:

- Query for the number of rows required to fill the query (`maxRows - rowsCount`)
- Query for whatever `limit` is specified as

Reply:
So I’m having a hard time writing a unit test that illustrates the issue that I see here 😅 I could probably do it, but I would have to modify the mockClient dependency method to give us a way to check how many rows are being fetched with every API call. What you have works, but in the case where a large `limit` is used and `maxRows` is undefined, it is likely to fetch way more rows than what is needed on the last API call.

Reply:
I think a solution is to always use maxRows in cursors (even when the `query` method instantiates the cursor. I'll incorporate that in my PR.